xds/internal/client/xds.go

```diff
@@ -522,8 +522,13 @@ func routesProtoToSlice(routes []*v3routepb.Route, logger *grpclog.PrefixLogger,
 				route.WeightedClusters[c.GetName()] = wc
 				totalWeight += w
 			}
-			if totalWeight != 100 && (wcs.GetTotalWeight() == nil || totalWeight != wcs.GetTotalWeight().GetValue()) {
-				return nil, fmt.Errorf("route %+v, action %+v, weights of clusters do not add up to total total weight, got: %v, want %v", r, a, wcs.GetTotalWeight().GetValue(), totalWeight)
+			// default TotalWeight https://github.com/grpc/grpc-go/issues/4432
```

Comment:
Nit: let's update the link to https://www.envoyproxy.io/docs/envoy/latest/api-v3/config/route/v3/route_components.proto.html#envoy-v3-api-field-config-route-v3-weightedcluster-total-weight, instead of the issue